### PR TITLE
documentation for TypeIs

### PR DESCRIPTION
**Pull request by chelseadz**
Fixes [#17156](https://github.com/python/mypy/issues/17156)

As requested in the issue, added documentation for the desired behaviour of TypeIs.

### CODE REVIEW

File: docs/source/type_narrowing.rst [hunk 1]
```diff
@@ -395,3 +395,157 @@ or rewrite the function to be slightly more verbose:
...
+
+TypeIs
+------
+
+Mypy should support TypeIs (:pep:`754`).
+
+A TypeIs function allows you to define custom type checks that
+can narrow the type of a variable in both the if and else branches
+of a conditional, similar to how the built-in isinstance() function works.
+
```
Comment (Hnasar) [inline docs/source/type_narrowing.rst:410]: You could mention something like 
> TypeIs is new in Python 3.13 — for use in older Python versions, use the backport from https://typing-extensions.readthedocs.io/en/latest/

File: docs/source/type_narrowing.rst [hunk 2]
```diff
@@ -395,3 +395,157 @@ or rewrite the function to be slightly more verbose:
...
+TypeIs functions can accept additional parameters beyond the first.
+The type narrowing applies only to the first argument.
+
+.. code-block:: python
+
+    from typing import Any, TypeVar, Type, reveal_type, TypeIs
+
+    T = TypeVar('T')
+
+    def is_instance_of(val: Any, typ: Type[T]) -> TypeIs[T]:
```
Comment (Hnasar) [inline docs/source/type_narrowing.rst:506]:
```suggestion
    from typing import Any, TypeVar, reveal_type, TypeIs

    T = TypeVar('T')

    def is_instance_of(val: Any, typ: type[T]) -> TypeIs[T]:
```
You don't need to use `typing.Type` any more

**Review by Hnasar — APPROVED** [review]
nice!

File: docs/source/type_narrowing.rst (hunk 2 above)
Comment (chelseadz) [reply docs/source/type_narrowing.rst]: okay! i applied the sugested changes

**Review by JukkaL — APPROVED** [review]
Thanks!

**Pull request merged**